src/main/java/crazypants/enderio/conduit/gui/ExternalConnectionContainer.java

```diff
@@ -193,6 +199,10 @@ public boolean canInteractWith(EntityPlayer entityplayer) {
 
   @Override
   public ItemStack slotClick(int par1, int par2, int par3, EntityPlayer par4EntityPlayer) {
+    ItemStack st = par4EntityPlayer.inventory.getItemStack();
+    if(st != null && st.getItem() == EnderIO.itemExtractSpeedUpgrade && st.getItemDamage() != 0) {
```

Comment:
These != 0 checks are not conducive to more types in the future. Would prefer an enum or something of the sort be created.